Toast notifications

florianluediger opened the pull request:
Progress for Issue #3 
Seems to not only work fine on Windows 10, but also on older versions like Windows 7.

The used library [Windows-10-Toast-Notifications](https://github.com/jithurjacob/Windows-10-Toast-Notifications) is not installable via pip for Python versions above 3.3 at the moment.(At least not via the requirements.txt) I have created a [pull request](https://github.com/jithurjacob/Windows-10-Toast-Notifications/pull/12) and we should wait until that is merged and the package is updated in the PyPi repository before merging this PR.

If you want to test the functionality right now, install the dependencies manually like this:
```
pip install pypiwin32
pip install win10toast --no-dependencies
```

Edit by @dtext:
Closes #3. We could merge this now, the feature is not usable in windows unless you install the dependencies manually as described above. The PR has been merged now, it's just a matter of time until pypi gets updated.

florianluediger committed:
Implement dispatching audio playback to a background thread so the execution does not get blocked.

florianluediger committed:
Implement Windows toast notification after the upload is done.
Although the used library is called win10toast, the notifications seem to work fine on older Windows versions.

dtext commented:
If you were kind enough to choose a less exclusive title for this PR, I would offer to implement this on Linux 😉

dtext committed:
Add Linux implementation of text notifications.

dtext committed:
Add our icon as default icon for toast notifications

dtext commented:
Regarding the dependency issue with the windows library: I will create a commit where I remove the dependency from the requirements file. Until your PR is merged, users of that feature will need to install it by hand. If the package on pypi gets updated, we can revert that commit. This way, we can merge this PR without breaking the requirements installation.

I will also create a reminder issue to revert that specific commit, so we don't forget.

dtext committed:
Remove win10toast dependency from requirements file

The library can't be installed using pip at the moment. There is a
pending PR to their repository that fixes that, so until that gets
merged, we need to remove this line, so as not to break requirements
installation.

This should be reverted as soon as possible.

dtext committed:
Minor fix in instantshare.default

dtext commented:
@denniseffing This should be ready for review now I think.
Edit: Merging now because you don't seem to have time to review, everything should be fine.
